src/estimators/tests/stratified_sketch_test.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2020 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for wfa_cardinality_estimation_evaluation_framework.estimators.stratified_sketch."""
+
+from absl import logging
+from absl.testing import absltest
+from absl.testing import parameterized
+
+import farmhash
+import numpy as np
+
+from wfa_cardinality_estimation_evaluation_framework.estimators import stratified_sketch
+from wfa_cardinality_estimation_evaluation_framework.estimators.exact_set import ExactMultiSet
+from wfa_cardinality_estimation_evaluation_framework.simulations import set_generator
+
+ONE_PLUS = '1+'
+
+class StratifiedTest(parameterized.TestCase):
+
+  def test_sketch_building_from_exact_multi_set(self):
+    max_freq = 3
+    expected = ExactMultiSet()
+    for k in range(max_freq + 2):
+        for i in range(k):
+            expected.add(k)
+
+    s = stratified_sketch.StratifiedSketch(sketch_list=[], max_freq=max_freq,
+          random_seed=np.random.RandomState(1))
+    s.init_from_exact_multi_set(expected, CardinalitySketch=ExactMultiSet)
+
+    self.assertLen(s.sketches.keys(), max_freq+1)
+    self.assertLen(s.sketches[ONE_PLUS], len(expected))
+
+    for id, freq in expected.ids().items():
+      freq_key = str(min(freq, max_freq))
+      self.assertTrue(id in s.sketches[freq_key])
+
+  def test_sketch_building_from_set_generator(self):
+    universe_size = 1000
+    set_sizes = [100] * 5
+    max_freq = 3
+
+    ind_gen = set_generator.IndependentSetGenerator(
+      universe_size=universe_size, set_sizes=set_sizes,
+      random_state=np.random.RandomState(1))
+    ind_gen_clone = set_generator.IndependentSetGenerator(
+      universe_size=universe_size, set_sizes=set_sizes,
+      random_state=np.random.RandomState(1))
```

Comment:
I would favor using a fake SetGenerator.  It allows you to control exactly what the resulting data structure should look like.  For an example, see lines 83-100 of simulations/tests/simulator_test.py.

Reply:
This is so helpful, was looking for this. Thanks!